crates/red_knot_module_resolver/src/resolver.rs

```diff
@@ -1140,4 +1386,207 @@ mod tests {
             system_path_to_file(&db, stdlib.join("functools.pyi"))
         );
     }
+
+    #[test]
+    fn editable_install_absolute_path() {
+        const SITE_PACKAGES: &[FileSpec] = &[("_foo.pth", "/x/src")];
+        let x_directory = [("/x/src/foo/__init__.py", ""), ("/x/src/foo/bar.py", "")];
+
+        let TestCase { mut db, .. } = TestCaseBuilder::new()
+            .with_site_packages_files(SITE_PACKAGES)
+            .build();
+
+        db.write_files(x_directory).unwrap();
+
+        let foo_module_name = ModuleName::new_static("foo").unwrap();
+        let foo_bar_module_name = ModuleName::new_static("foo.bar").unwrap();
+
+        let foo_module = resolve_module(&db, foo_module_name.clone()).unwrap();
+        let foo_bar_module = resolve_module(&db, foo_bar_module_name.clone()).unwrap();
+
+        assert_eq!(
+            Some(foo_module.file()),
+            system_path_to_file(&db, SystemPathBuf::from("/x/src/foo/__init__.py"))
+        );
```

Comment:
Nit: I think I prefer comparing the paths. It gives nicer error messages (1 != 2)
```suggestion
        assert_eq!(
            foo_module.file().path,
            FilePath::System(SystemPathBuf::from("/x/src/foo/__init__.py"))
        );
```